pkg/roll/fields/update.go

```diff
@@ -0,0 +1,57 @@
+package fields
+
+import (
+	"github.com/square/p2/pkg/rc/fields"
+)
+
+// An Update (roll.Update, borrowed from kubectl's "rolling-update" command)
+// represents a transition from one replication controller to another. The IDs
+// represent the two RCs involved in the transition.
+type Update struct {
+	// The old replication controller that will be releasing control. At the end
+	// of the Update, this RC will be deleted.
+	OldRC fields.ID
+	// Thew new replication controller that will be taking over. Updates can be
+	// uniquely identified by their new RC's ID.
+	NewRC fields.ID
+	// When the new RC's replica count equals this number, the Update is
+	// considered to be complete. This should be at least equal to the minimum.
+	DesiredReplicas int
+	// During the course of the update, pods may be restarted or removed. This
+	// number specifies the minimum number of replicas that must be alive
+	// during the entire Update.
+	//
+	// The acceptable value of Minimum is subject to several constraints:
+	//   - the update will not even start until the minimum is satisfied, so if
+	//     the current number of live nodes is than the minimum, it will block
```

Comment:
yep there's a debug log around line 150 of run_update.go to indicate that it's blocked